lib/sqlalchemy/engine/reflection.py

```diff
@@ -768,14 +768,12 @@ def reflect_table(
             if isinstance(table_name, str):
                 table_name = table_name.decode(dialect.encoding)
 
-        found_table = False
+        found_table = True if table.name in self.engine.table_names() else False
```

Comment:
`eninge.table_names()` is deprecated.

Please use the inspector `has_table` method.

Also I think it makes more sense to add this control only if found_table = False after the loop, avoiding a query in most cases

Reply:
thanks @CaselIT for reviewing, I'll look up `has_table` thanks for letting me know `.table_names()` is deprecated. 
should we be having `found_table` be set in a loop, wondering this maybe redundant albeit harmless, considering tables where cols are 1000 it is unnecessarily setting it to the very value it already is. 

Nevertheless I can rather have this `has_table` check setting `found_table` after the loop/before raising exc.

Reply:
the reasoning is that I would think the use case for tables with no columns is really small, it's the first time it came up in 16+ years, so we can avoid the additional has_table query most of the times.

But it's probably not too important, so it's fine also to leave as is, just replacing the table_names with has_table.

> thanks for letting me know `.table_names()` is deprecated.

well it does raise a deprecation warning in your output here https://github.com/sqlalchemy/sqlalchemy/issues/7117

Reply:
@CaselIT sorry I overlooked. Nevertheless, I've replaced the deprecated `table_names()` with `has_table(table.name)`.
Secondly,
> Also I think it makes more sense to add this control only if found_table = False after the loop, avoiding a query in most cases

I believe the `found_table` currently, is being done with, bit redundantly. Ideally, we should check for `found_table` at once using this fix, and **right away raise the NoSuchTableError exception** should it be False. Thereafter, we can do the needful on reflecting columns. Let me know 👍

Reply:
makes sense, we could as well do `if not self.has_table(table): raise ...`

Reply:
done. 
- so removed redundant `found_table`. 
- made referring/using `self.has_table(table.name)` more compact, as suggested above.

Reply:
@CaselIT apparently it is failing for all python2.7 envs and cancelled operations for `3.9-cext` and `3.9-nocext` envs), while the error, more than breaking changes rather looks like a NoSuchTableError being thrown for tables like "dingalings", "foob", "a" etc:
```
    metadata._remove_table(name, schema)
  File "/home/runner/work/sqlalchemy/sqlalchemy/.tox/github-cext/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 72, in __exit__
    with_traceback=exc_tb,
  File "/home/runner/work/sqlalchemy/sqlalchemy/.tox/github-cext/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 602, in __new__
    table._init(name, metadata, *args, **kw)
  File "/home/runner/work/sqlalchemy/sqlalchemy/.tox/github-cext/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 682, in _init
    resolve_fks=resolve_fks,
  File "/home/runner/work/sqlalchemy/sqlalchemy/.tox/github-cext/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 717, in _autoload
    _extend_on=_extend_on,
  File "/home/runner/work/sqlalchemy/sqlalchemy/.tox/github-cext/lib/python2.7/site-packages/sqlalchemy/engine/reflection.py", line 785, in reflect_table
    raise exc.NoSuchTableError(table.name)
NoSuchTableError: dingalings
```
Are these tables present at first, in the PostgresSQL on the server, being referred to (appropriate schema)? Wondering if I could get a step closer debugging this, ***interactively***. Doesn't look like a break due to changes in this PR.

Reply:
It seems schema is missing form the has_table call

Reply:
ok, yes perhaps the schema being picked up was a `None` owing to the definition:
```
def has_table(self, table_name, schema=None):
        """Return True if the backend has a table of the given name.
```

have fixed it as calling has_table(table.name, schema) since the variable `schema` is set well above and being used (e.g for getting/reflecting columns). Will keep an eye as the pipeline/workflow triggers and watch of for failing envs.

Reply:
thanks I'll make it execute